testing/dart/text_test.dart

```diff
@@ -241,4 +241,21 @@ void main() {
   testTextRange();
   testLoadFontFromList();
   testFontFeatureClass();
+
+  test('Invalid UTF-16 does not cause crash', () async {
+    // First half of this UTF32 code point is not a valid UTF16 code point.
+    final String invalidString = '💩'.substring(0, 1);  // i.e. codeUnitAt(0)
+
+    final ParagraphBuilder builder = ParagraphBuilder(ParagraphStyle());
+    builder.addText(invalidString);
+    final Paragraph paragraph = builder.build();
+    paragraph.layout(const ParagraphConstraints(width: double.infinity));
+
+    final PictureRecorder recorder = PictureRecorder();
+    final Canvas canvas = Canvas(recorder);
+    // This must not crash.
```

Comment:
Is there any notion of what should happen here beyond not crashing?

Reply:
I suppose we could try to assert whether it actually draws a replacement character or an empty string... That would require a golden test though, which would be easier to write in the framework.

Reply:
The intention of this test though is to be a fuzzing test though

Reply:
> The intention of this test though is to be a fuzzing test though

That's fair. Not crashing is the change in behavior for this PR, so that's what it makes sense to test.